### DISCUSSION TITLE
Why was the useVirtualList function deprecated?

### DISCUSSION THREAD

**Discussion by mcfarljw** [vueuse/vueuse#2683] [General]
I noticed recently that `useVirtualList` (https://vueuse.org/core/usevirtuallist/#usevirtuallist) was deprecated and now recommends using `vue-virtual-scroller` (https://github.com/Akryum/vue-virtual-scroller). I was a bit surprised to see this because I've always found the latter to be buggy, overly complicated, and lacking in TypeScript support.

<img width="624" alt="Screenshot 2023-01-27 at 11 49 27 AM" src="https://user-images.githubusercontent.com/4167341/215145516-5115d0ab-d903-4615-9256-6c8bd42223ca.png">

I've also noticed in https://github.com/elk-zone/elk which has a number of overlapping contributors with this project that it's currently disabled with at least one open issue for it which is a bit of a red flag for me indicating it's not too stable.

https://github.com/elk-zone/elk/blob/3e365fc30a129a07349d53f3d315b094fa125edb/components/common/CommonPaginator.vue#L11 currently has it defaulting to off.

https://github.com/elk-zone/elk/issues/629 is one of the issue I've experienced with more complex use cases that have been around since at least 2019.

I'm more than willing to switch over to it but wanted to voice my opinion and see if there were any additional details regarding the decision as I think both solutions can coexist.

### UPDATE
The deprecation notice was removed via https://github.com/vueuse/vueuse/pull/2756 with an updated note about using the `vue-virtual-scroller` for additional functionality if needed.

**Comment by nockstarr** [comment]
I also opt for `useVirtualList` to stay. Its simple to use and solves many use cases with ease. It does not have to be the most fully featured component out there... We can still link to `vue-virtual-scroller` in the docs.

`useVirtualList` also seems to be referenced quite a lot in open repos
https://github.com/search?q=vueuse+useVirtualList&type=code

**Comment by JessicaSachs** [comment]
I echo @nockstarr's suggestion: I would like to link out to `vue-virtual-scroller` and **suggest its usage if people are looking for more features**.

The reason I contributed to `useVirtualList` over a year ago is because of the quirks and bugs I found in `vue-virtual-scroller`. I tried and tried all of the different options, but it couldn't handle my usecase. Eventually, I submitted a PR to `useVirtualList` and that's what we used for Cypress's own UI.

While I understand that it's difficult to get Virtual Scrollers right and `useVirtualList` adds to the maintenance burden and number of functions within VueUse, I don't believe `vue-virtual-scroller` can replace it without significant burden to the community.

**Considerations:**
1. It has two separate packages for Vue 2 and Vue 3.
    1. These two packages need to be maintained + bug fixes must be backported by the maintainer.
    2. It will likely lose Vue 2 support sooner than `useVirtualList` will.
2. It does not have the documentation, TypeScript support, stability, or number of contributors that VueUse and `useVirtualList` does.

**Comment by wheatjs** [comment]
These are some good points, I like the suggestion of linking out to `vue-virtual-scroller` for more features. If someone would like to make a PR to remote the deprecation notice I'll be happy to approve it
